lib/hets/hets_options.rb

```diff
@@ -46,9 +47,19 @@ def remove_nil_fields
     end
 
     def prepare_url_catalog
-      @options[:'url-catalog'].try(:compact!)
-      if @options[:'url-catalog'].blank?
-        @options.delete(:'url-catalog')
+      if @options[:'url-catalog'].is_a?(Array)
+        @options[:'url-catalog'].try(:compact!)
+        if @options[:'url-catalog'].blank?
+          @options.delete(:'url-catalog')
+        else
+          @options[:'url-catalog'] = @options[:'url-catalog'].join(',')
+        end
+      end
+    end
+
+    def prepare_access_token
+      if @options[:'access-token'].is_a?(AccessToken)
```

Comment:
Yeah, but `to_s` is basically a noop on Strings, so you could just skip the altogether, couldn't you?